controllers/kserve_customcacert_controller.go

```diff
@@ -0,0 +1,190 @@
+		if existingOdhCustomCertData == odhCustomCertData {
+			log.Info(fmt.Sprintf("No updates required for KServe custom cert ConfigMap (%s) as the data matches the opendatahub global cert ConfigMap (%s)", kserveCustomCACertConfigMapName, odhGlobalCACertConfigMapName))
+			return nil
+		}
+		log.Info(fmt.Sprintf("Updating KServe custom cert ConfigMap due to changes in the opendatahub global cert ConfigMap (%s)", odhGlobalCACertConfigMapName))
+		if err := r.Update(ctx, newCaCertConfigMap); err != nil {
+			return err
+		}
+		if err := r.deleteStorageSecret(ctx, configmap.Namespace); err != nil {
+			log.Error(err, "Failed to delete the storage-config secret to update the custom cert")
+			return err
+		}
+		log.V(1).Info("Deleted the storage-config Secret to update the custom cert")
+	}
+
+	return nil
+}
+
+// This section is intended for regenerating StorageSecret using new data.
+func (r *KServeCustomCACertReconciler) deleteStorageSecret(ctx context.Context, namespace string) error {
+	foundStorageSecret := &corev1.Secret{}
+
+	err := r.Get(ctx, types.NamespacedName{
+		Name:      constants.DefaultStorageConfig,
+		Namespace: namespace,
+	}, foundStorageSecret)
+
+	if err == nil {
+		err = r.Delete(ctx, foundStorageSecret)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func checkOpenDataHubGlobalCertCAConfigMapName(objectName string) bool {
+	return objectName == odhGlobalCACertConfigMapName
+}
+
+// reconcileOpenDataHubGlobalCertConfigMap filters out all ConfigMaps that are not the OpenDataHub global certificate ConfigMap.
+func reconcileOpenDataHubGlobalCACertConfigMap() predicate.Predicate {
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			objectName := e.Object.GetName()
+			return checkOpenDataHubGlobalCertCAConfigMapName(objectName)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			objectName := e.Object.GetName()
+			return checkOpenDataHubGlobalCertCAConfigMapName(objectName)
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			objectName := e.Object.GetName()
+			return checkOpenDataHubGlobalCertCAConfigMapName(objectName)
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			objectName := e.ObjectNew.GetName()
+			return checkOpenDataHubGlobalCertCAConfigMapName(objectName)
+		},
+	}
+}
+
+func (r *KServeCustomCACertReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	// Initialize logger format
+	log := r.Log.WithValues("ConfigMap", req.Name, "namespace", req.Namespace)
+
+	configmap := &corev1.ConfigMap{}
+	err := r.Get(ctx, req.NamespacedName, configmap)
+	if err != nil && apierrs.IsNotFound(err) {
+		log.Info("Opendatahub global cert ConfigMap not found")
+		configmap.Namespace = req.Namespace
+	} else if err != nil {
+		log.Error(err, "Unable to fetch the ConfigMap")
+		return ctrl.Result{}, err
+	}
+
+	err = r.reconcileConfigMap(configmap, ctx)
```

Comment:
we shoud pass the `log` instace that we created above into `reconcileConfigMap` method. I am asking it because in above line we created a new log instance by appending the ConfigMap name and namespace details. So we pass same reference then every log would print these details by default.

Reply:
updated.